packages/widgets/src/blocks/legacy-widget/edit/preview.js

```diff
@@ -11,9 +11,19 @@ import { addQueryArgs } from '@wordpress/url';
 import { useState } from '@wordpress/element';
 import { Placeholder, Spinner, Disabled } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
 
 export default function Preview( { idBase, instance, isVisible } ) {
 	const [ isLoaded, setIsLoaded ] = useState( false );
+	const { adminUrl } = useSelect( ( select ) => {
```

Comment:
The adminUrl doesn't make sense as a block editor setting. The block-editor package is a generic package that can be used to initialize any kind of block editor, even WP agnostic. That setting seem more suited for the widgets screen settings it seems.

Reply:
Hmm, I'm not sure I understand. Does it mean that the `settings` prop in `<BlockEditorProvider>` is only supposed to contain fields defined in [`SETTINGS_DEFAULT`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/README.md#settings_defaults)? I thought it were just a settings context that users can pass their user-defined fields to every component via React Context. The `block-editor` package doesn't have to be aware of these user-defined fields.

Instead of passing `adminUrl` to every editor from the backend, we can only register it in the widgets editors, but I would still imagine using block editor's `settings` as the canonical way. Is the alternative creating a separate context to pass those values?

Reply:
The settings passed to the `edit-post` package from the backend are splitted between:

 - Settings kept in the edit-post store because are specific to edit-post
 - Settings passed down to `BlockEditorProvider` because they change the behavior the block editor. 

`BlockEditorProvider` is not meant to be used as a generic settings bag. the simplest solution I can think of is to use React Context for that or a store like edit-post/editor.

At least that's how it is currently.